tests/foreman/api/test_convert2rhel.py

```diff
@@ -0,0 +1,323 @@
+    subs = entities.Subscription(organization=module_org).search(
+        query={'search': f'{DEFAULT_SUBSCRIPTION_NAME}'}
+    )
+    assert subs
+    return create_activation_key(module_org, module_lce, module_promoted_cv, subs[0].id)
+
+
+def enable_rhel_repos(org, repos):
+    rh_repos = []
+    for repo in repos:
+        rh_repo_id = enable_rhrepo_and_fetchid(
+            basearch=DEFAULT_ARCHITECTURE,
+            org_id=org.id,
+            product=REPOS[repo]['product'],
+            repo=REPOS[repo]['name'],
+            reposet=REPOS[repo]['reposet'],
+            releasever=REPOS[repo]['releasever'],
+        )
+        # Sync step because repo is not synced by default
+        rh_repo = entities.Repository(id=rh_repo_id).read()
+        rh_repo.sync(timeout=2500)
+        rh_repos.append(rh_repo)
+    return rh_repos
+
+
+@pytest.fixture(scope='module')
+def enable_rhel7_subscriptions(module_org, manifest):
+    """Enable and sync RHEL7 server rpms repo"""
+    return enable_rhel_repos(module_org, ['rhel7'])
+
+
+@pytest.fixture(scope='module')
+def enable_rhel8_subscriptions(module_org, manifest):
+    """Enable and sync RHEL8 baseos and appstream repos"""
+    return enable_rhel_repos(module_org, ['rhel8_bos', 'rhel8_aps'])
+
+
+@pytest.fixture
+def centos_host(
+    request, default_sat, module_org, smart_proxy_location, module_promoted_cv, module_lce, version
+):
+    """Deploy and register Centos host"""
+    rh_repos = request.getfixturevalue(f"enable_rhel{version}_subscriptions")
+    conf = {
+        "workflow": "deploy-centos",
+        "deploy_scenario": "centos",
+        "deploy_rhel_version": version,
+    }
+    with VMBroker(**conf, host_classes={'host': ContentHost}) as host:
+        host.install_katello_ca(default_sat)
+        # updating centos packages on CentOS 8 is necessary for conversion
+        if version == '8':
+            host.execute("yum update -y centos-*")
+        repo = create_repo(
+            module_org, settings.repos.convert2rhel.convert_to_rhel_repo.format(version)
+        )
+        cv = update_cv(module_promoted_cv, module_lce, rh_repos + [repo])
+        c2r_sub = entities.Subscription(organization=module_org, name=repo.product.name).search()[0]
+        act_key = create_activation_key(module_org, module_lce, cv, c2r_sub.id)
+        register_host(act_key, module_org, smart_proxy_location, host.hostname)
+        yield host
+        # close ssh session before teardown, because of reboot in conversion it may cause problems
+        host.close()
+
+
+@pytest.fixture
+def oracle_host(
+    request,
+    default_sat,
+    module_org,
+    smart_proxy_location,
+    module_promoted_cv,
+    module_lce,
+    ssl_cert,
+    version,
+):
+    """Deploy and register Oracle host"""
+    rh_repos = request.getfixturevalue(f"enable_rhel{version}_subscriptions")
+    conf = {
+        "workflow": "deploy-oracle-linux",
```

Comment:
Same as above - `conf/content_host.yaml` structure should be updated to include non-rhel settings for workflow names and kwargs.